Support create RM storage Account with Kind "Storagev2"

blueww opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
Support create RM storage Account with Kind "Storagev2"
Also need to upgrade SRP SDK to 7.0. Since there's a breaking change in SRP SDK API for NetworkACL->NetworkRuleSet, PowerShell also has code change on it. But since Powershell has wrappered all interface for NetworkRuleSet, PowerShell don't have breaking on it.
---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [Y ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [Y ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [Y ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [Y ] Title of the pull request is clear and informative.
- [Y ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [Y ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [Y ] Pull request includes test coverage for the included changes.
- [Y ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ Y] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [Y ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [Y ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [Y ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [Y ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

cormacpayne commented:
@azuresdkci test this please

cormacpayne commented on src/ResourceManager/Storage/Commands.Management.Storage/Commands.Management.Storage.csproj:
```diff
@@ -67,10 +67,13 @@
       <HintPath>..\..\..\packages\Microsoft.Data.Services.Client.5.8.2\lib\net40\Microsoft.Data.Services.Client.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL" />
```
@blueww you can remove the references added in this file (`Microsoft.Rest.*`, `Newtonsoft.Json`) since they are already referenced by the `Common.Dependencies.targets` file

cormacpayne commented on src/ResourceManager/Storage/Commands.Management.Storage/packages.config:
```diff
@@ -1,6 +1,9 @@
   <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.Storage" version="6.5.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.Storage" version="7.1.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.10" targetFramework="net452" />
```
@blueww same comment about removing the added references since they are already referenced from a common project

blueww commented:
@cormacpayne 

For the breaking change, it's for New/Set-AzureStorageAccount cmdelts output has type "Microsoft.Azure.Management.Storage.Models.StorageAccount", and NetworkAcls is remove from the type.

But actually, the real return type of New/Set-AzureStorageAccount is "PSStorageAccount" (Yes, PSStorageAccount is already defined in PSH code, and used as output type of account cmdlets), you can see both cmdelts use "WriteStorageAccount(storageAccount)" to write output, and actually output Type is "PSStorageAccount":

```
        protected void WriteStorageAccount(StorageModels.StorageAccount storageAccount)
        {
            WriteObject(PSStorageAccount.Create(storageAccount, this.StorageClient));
        }
```

So I have correct the output type in New/Set-AzureStorageAccount to "PSStorageAccount". This will show a breaking change in StaticAnalysis, but it's actually not a breaking, since the output of the 2 cmdelts not change (only the output section in help change). So I have added the 2 output type change into Exception of breaking change.

I have fixed all other comments from you.
Let me know if you have any concern.
Thanks!

cormacpayne commented:
@blueww the issue still remains: the `NetworkAcls` property is no longer accessible from the returned object of `New/Set-AzureRmStorageAccount`. You will need to add this property to the `PSStorageAccount` type so users are still able to use it in their scripts. Once you add this property, you can add the `Obsolete` attribute on it and let users know it will be removed in an upcoming breaking change release.

cormacpayne commented:
@blueww one other thing, please make sure to update the `ChangeLog.md` file located in `src/ResourceManager/Storage` witha snippet about the changes you are making in this PR

blueww commented:
@cormacpayne 
The Change log is added ,and on-demand test updated on https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/196/.

For the `NetworkACLs`, I might don't clarify it clearly. 
Actually the original output type of the 2 effected cmdelts `New/Set-AzureRmStorageAccount `are `PSStorageAccount `, you can see it in the code, all output storage Account will conver to `PSStorageAccount `, then output. So there's no need to add `NetworkACLs`, since the original output don't have `NetworkACLs`.

The original breaking is come from the output Type declared in the begin of cmdlet code is wrong. It 's set to `OutputType(typeof(StorageModels.StorageAccount))`, but the actually output type is `PSStorageAccount `, so I just correct the output type for the 2 cmdelts to OutputType(typeof(PSStorageAccount)), this make the real output type and the output type declared in code aligned. This fix is not a breaking, since it not change the output type, but just correct the Output Section in help *.md file.

cormacpayne commented:
@blueww thanks for the clarification ☺️ everything looks good
